central/processlisteningonport/store/postgres/store.go

```diff
@@ -0,0 +1,806 @@
+// Code generated by pg-bindings generator. DO NOT EDIT.
+
+// XXX: There are custom bits applied on top of the generated code. Everything
+// what comes around GetPLOPForDeployment is manually written.
+
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/logging"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/sync"
+	"gorm.io/gorm"
+)
+
+const (
+	baseTable = "process_listening_on_ports"
+
+	batchAfter = 100
+
+	// using copyFrom, we may not even want to batch.  It would probably be simpler
+	// to deal with failures if we just sent it all.  Something to think about as we
+	// proceed and move into more e2e and larger performance testing
+	batchSize = 10000
```

Comment:
Based on what all this thing is doing, I would probably reduce this and the deleteBatchSize.

Reply:
Can you elaborate pls, why reduce?

Reply:
Just something to be mindful of when you get around to running some scale and performance type tests.  It may actually be a more likely problem for indicators than PLOP due to the additional multi-column unique index slowing down writes.